MantidQt/API/src/QwtRasterDataMDNonOrthogonal.cpp

```diff
@@ -0,0 +1,172 @@
+#include "MantidQtAPI/QwtRasterDataMDNonOrthogonal.h"
+#include "MantidQtAPI/NonOrthogonal.h"
+
+namespace MantidQt {
+namespace API {
+
+using namespace Mantid;
+using namespace Mantid::API;
+
+QwtRasterDataMDNonOrthogonal::QwtRasterDataMDNonOrthogonal()
+    : m_lookPoint(nullptr) {
+  m_skewMatrix[0] = 1.0;
+  m_skewMatrix[1] = 0.0;
+  m_skewMatrix[2] = 0.0;
+  m_skewMatrix[3] = 0.0;
+  m_skewMatrix[4] = 1.0;
+  m_skewMatrix[5] = 0.0;
+  m_skewMatrix[6] = 0.0;
+  m_skewMatrix[7] = 0.0;
+  m_skewMatrix[8] = 1.0;
+
+  m_missingHKLdim = 0;
+}
+
+QwtRasterDataMDNonOrthogonal::~QwtRasterDataMDNonOrthogonal() {
+  delete[] m_lookPoint;
+}
+
+//-------------------------------------------------------------------------
+/** Return the data value to plot at the given position
+*
+* @param x :: position in coordinates of the MDWorkspace
+* @param y :: position in coordinates of the MDWorkspace
+* @return signal to plot
+*/
+double QwtRasterDataMDNonOrthogonal::value(double x, double y) const {
+  if (!m_ws)
+    return 0;
+
+  // Generate the vector of coordinates, filling in X and Y
+  for (size_t d = 0; d < m_nd;
+       d++) { // m_nd is number dimensions in the workplace
+    if (d == m_dimX)
+      m_lookPoint[d] = static_cast<coord_t>(x);
+    else if (d == m_dimY)
+      m_lookPoint[d] = static_cast<coord_t>(y);
+    else
+      m_lookPoint[d] = m_slicePoint[d];
+  }
+
+  // Transform the lookpoint to the coordinate of the workspace
```

Comment:
Maybe we also want to refactor this, since it does the same thing several times. The indirection could cause a performance issue since this function is called many times. So we should probably profile this to see if refactoring causes a performance hit